src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -2502,6 +2502,16 @@ bool QgsPostgresProvider::addFeatures( QgsFeatureList &flist, Flags flags )
         {
           v = paramValue( value.toString(), defaultValues[ i ] );
 
+          // the conversion functions expects the list as a string, so convert it
+          if (value.type() == QVariant::StringList)
+          {
+              v = "{\"" + value.toStringList().join("\",\"") + "\"}";
```

Comment:
```suggestion
              v = QStringLiteral( "{\"" ) + value.toStringList().join( QStringLiteral( "\",\"") ) + QStringLiteral( "\"}" );
```

Reply:
This lacks logic to escape characters like double-quote. I'm wondering if you shouldn't use the quotedValue() method for this case and the below one.

A new test in tests/src/python/test_provider_postgres.py would be welcome

Reply:
I started the tests in c++ in testqgslistwidget.cpp as I thought most of this would relate to list widget only, so added that for now, but I can see the potential benefit of having it further up the chain so I could translate into python if it's considered worthwhile to do so.

I didn't use quotedValue() in the end as I think the array syntax is slightly different to normal strings, and so all that I think is needed is to escape double quotes within the string.